judge/models/profile.py

```diff
@@ -188,7 +188,9 @@ def generate_api_token(self):
     generate_api_token.alters_data = True
 
     def generate_scratch_codes(self):
-        codes = [pyotp.random_base32(length=16) for i in range(settings.DMOJ_SCRATCH_CODES_COUNT)]
+        def generate_scratch_code():
+            return "".join(secrets.choice(string.ascii_uppercase) for _ in range(16))
```

Comment:
The alphabet size for this new scratch code generator is actually smaller, since it used to be base32 and now it's only the 26 ASCII uppercase letters. There is no need to decrease security in this case.

Reply:
Fixed.